framework/src/media/audio/audio_manager.c

```diff
@@ -212,23 +267,45 @@ audio_manager_result_t set_audio_volume(uint16_t volume)
 	return AUDIO_MANAGER_SUCCESS;
 }
 
-int get_audio_volume(void)
+audio_manager_result_t set_input_audio_volume(uint16_t volume)
 {
-	int ret;
-	int volume = -1;
-	struct pcm *pcm;
+	audio_config_t audio_config = g_audio_in_cards[g_actual_audio_in_card_id].config;
 
-	if ((ret = get_active_audio_device_pcm(&pcm, OUTPUT)) != AUDIO_MANAGER_SUCCESS) {
-		return ret;
+	if (volume > AUDIO_DEVICE_MAX_VOLUME) {
+		volume = AUDIO_DEVICE_MAX_VOLUME;
 	}
 
-	ret = ioctl(pcm_get_file_descriptor(pcm), AUDIOIOC_GETVOLUME, (unsigned long)&volume);
-	if (ret < 0) {
-		auddbg("Fail to ioctl AUDIOIOC_GETVOLUME, ret = %d\n", ret);
-		return AUDIO_MANAGER_DEVICE_FAIL;
+	audio_config.cur_volume = volume * audio_config.max_volume / AUDIO_DEVICE_MAX_VOLUME;
+
+	return set_audio_volume(INPUT, audio_config.cur_volume);
+}
+
+audio_manager_result_t set_output_audio_volume(uint16_t volume)
+{
+	audio_config_t audio_config = g_audio_out_cards[g_actual_audio_out_card_id].config;
+
+	if (volume > AUDIO_DEVICE_MAX_VOLUME) {
+		volume = AUDIO_DEVICE_MAX_VOLUME;
 	}
 
-	return volume;
+	audio_config.cur_volume = volume * audio_config.max_volume / AUDIO_DEVICE_MAX_VOLUME;
```

Comment:
If current volume is same as requested one, just return needed.

Reply:
Added.